### DISCUSSION TITLE
Babel seems to include incorrect core-js modules

### DISCUSSION THREAD

**Discussion by otakustay** [babel/babel#15360] [Q&A] [answered]
Input code:

```js
"".replaceAll(/abc/g, "xyz")
```

Using options:

```json
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": "ie 10",
        "useBuiltIns": "usage",
        "corejs": 3
      }
    ]
  ]
}
```

Current output:

```js
"use strict";

require("core-js/modules/es.regexp.exec.js");
require("core-js/modules/es.string.replace.js");
require("core-js/modules/esnext.string.replace-all.js");
"".replaceAll(/abc/g, "xyz");
```

The prolem is `core-js/modules/esnext.string.replace-all.js` module, I tried to use `core-js-compat` to determine which core-js modules should be included:

```js
const compat = require('core-js-compat');

const {list} = compat({targets: 'ie 10'});
console.log(list.filter(v => v.includes('replace-all')));
```

It reports `[ 'es.string.replace-all', 'esnext.symbol.replace-all' ]` which fails to match the one babel given.

This case appers to me because I'm trying to "prebundle" required core-js modules in a specific build stage, and found some are not included in my bundle.

Just wonder how babel determines required core-js modules? Isn't it comes from `core-js-compat`?

**Comment by JLHwung** [comment, marked as answer]
Please specify the exact `core-js` version you are using, otherwise Babel will assume you are using `3.0`, which does not come with `es.string.replace-all`:

```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": "ie 10",
        "useBuiltIns": "usage",
        "corejs": "3.27"
      }
    ]
  ]
}
```
See also https://babeljs.io/docs/en/babel-preset-env#corejs
